src/native/corehost/fxr/standalone/hostpolicy_resolver.cpp

```diff
@@ -242,11 +242,15 @@ bool hostpolicy_resolver::try_get_dir(
     pal::string_t resolved_deps = get_deps_file(is_framework_dependent, app_candidate, specified_deps_file, fx_definitions);
 
     // Resolve hostpolicy version out of the deps file.
-    pal::string_t version = resolve_hostpolicy_version_from_deps(resolved_deps);
-    if (trace::is_enabled() && version.empty() && pal::file_exists(resolved_deps))
+    pal::string_t version;
+    if (pal::file_exists(resolved_deps))
     {
-        trace::warning(_X("Dependency manifest %s does not contain an entry for %s"),
-            resolved_deps.c_str(), _STRINGIFY(HOST_POLICY_PKG_NAME));
+        version = resolve_hostpolicy_version_from_deps(resolved_deps);
+        if (trace::is_enabled() && version.empty())
+        {
+            trace::warning(_X("Dependency manifest %s does not contain an entry for %s"),
+                resolved_deps.c_str(), _STRINGIFY(HOST_POLICY_PKG_NAME));
+        }
     }
 
     // Check if the given version of the hostpolicy exists in servicing.
```

Comment:
Can probably move this under the check above, since we'll never find it if the deps didn't exist or we didn't get a version to look for.